samples/snippets/grant_tag_template_user_role.py

```diff
@@ -0,0 +1,67 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+def grant_tag_template_user_role(override_values):
+    """Grants a user the Tag Template User role for a given template."""
+    # [START data_catalog_grant_tag_template_user_role]
+    from google.cloud import datacatalog_v1
+    from google.iam.v1 import iam_policy_pb2 as iam_policy
+    from google.iam.v1 import policy_pb2
+
+    datacatalog = datacatalog_v1.DataCatalogClient()
+
+    # TODO: Set these values before running the sample.
+    project_id = "project_id"
+    tag_template_id = "existing_tag_template_id"
+    member_id = "user:super-cool.test-user@gmail.com"
```

Comment:
Should we link users to something that describes the format of IAM member patterns?

Reply:
Will link to https://cloud.google.com/iam/docs/reference/rest/v1/Policy?hl=en#binding